bigbite/themer
# Enable adjustment of 'settings' values

## Description



Fixes #94 

This PR adds the feature to enable adjustment of 'settings' values in the themer editor. Suitable individual components have been built for each setting value. This could do with a review from a UI/UX point of view.

Settings are accessed using a 'settings' button to the right of each section. Settings are available site wide and for each individual block.

![image](https://github.com/user-attachments/assets/6444504e-9528-4258-9faf-afc10888fdc0)

Each component is currently wrapped in a panel for convenience as there's a lot of options available. We can adjust this to something more suitable if required.

https://github.com/user-attachments/assets/59ab9cd2-d946-4c4f-8396-c491838fa28f

I've added in repeating functionality where required (e.g. color palettes) and the option to edit or delete any created values.

![image](https://github.com/user-attachments/assets/070b7707-8a95-4591-802c-0cd5f7bb80c7)


## Change Log



- Adds in settings nav option
- Adds in suitable components for each setting value

## Steps to test



Open up themer and and click the new settings button available next to 'Site' and each block.
Test each component and check the values can be updated with the relevant data type and saved correctly.
Test the new options are now available in the post editor.

## Checklist:
